psh/cat/cat.c

```diff
@@ -45,38 +46,56 @@ int psh_cat(int argc, char **argv)
 	char *buff;
 	size_t ret, len, wrote;
 	int c, i, retval = EXIT_SUCCESS;
+	struct stat sbuff;
 
-	while ((c = getopt(argc, argv, "h")) != -1) {
+	c = getopt(argc, argv, "h");
+	while (c != -1) {
```

Comment:
Less error prone would be to use a single getopt call, maybe sth like the following:
(I'd replace boolean type switch-case because of MISRA Rule 16.7)
```suggestion
	for (;;) {
		c = getopt(argc, argv, "h");
		if (c == -1) {
			break;
		}

		psh_cat_help(argv[0]);
		return (c == 'h') ? EXIT_SUCCESS : EXIT_FAILURE;
	}
```